Neuro/Vision/VisionHandler.cs

```diff
@@ -0,0 +1,258 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Il2CppInterop.Runtime.InteropTypes.Arrays;
+using Neuro.Recording.DataStructures;
+using Neuro.Vision.DataStructures;
+using Reactor.Utilities.Attributes;
+using UnityEngine;
+
+namespace Neuro.Vision;
+
+// TODO: Refactor this entire class
+[RegisterInIl2Cpp]
+public class VisionHandler : MonoBehaviour
+{
+    public VisionHandler(IntPtr ptr) : base(ptr) { }
+
+    public Vector2 DirectionToNearestBody { get; set; }
+    public readonly Dictionary<byte, PlayerRecord> PlayerRecords = new(); // TODO: Store this data using a monobehaviour on the player
+
+    // TODO: Handle players disconnecting
+    private readonly List<DeadBody> deadBodies = new();
+    private readonly Dictionary<byte, PlayerControl> playerControls = new(); // TODO: Use GameData.GetPlayerById
+    private readonly Dictionary<PlayerControl, LastSeenPlayer> playerLocations = new(); // TODO: Don't use PlayerControls as dictionary keys,
```

Comment:
The conversation for this was resolved despite the dictionary not switching to using `PalyerID`?